README.md

```diff
@@ -0,0 +1,684 @@
+  
+# 1. Introduction
+
+UCAN Delegation is a certificate capability system with runtime-extensibility, ad hoc caveats, cacheability, and focused on ease of use and interoperability. Delegations act as a proofs for [UCAN Invocation]s.
+
+Delegation provides a way to "transfer authority without transferring cryptographic keys". As an authorization system, it is more interested in "what" can be done than a list of "who can do what". For more on how Delegation fits into UCAN, please refer to the [high level spec][UCAN].
+
+# 2. Delegation (Envelope)
+
+Delegations MUST include a signature that validates against the `iss` DID. A Delegation Payload on its own MUST NOT be considered a valid Delegation.
+
+| Field   | Type        | Required | Description                                  |
+|---------|-------------|----------|----------------------------------------------|
+| `ucd`   | `&Payload`  | Yes      | The CID of the [Delegation Payload][Payload] |
+| `sig`   | `Signature` | Yes      | The `iss`'s [Signature] over the `ucd` field |
+
+
+# 3. Delegation Payload
+
+The payload MUST describe the authorization claims, who is involved, and its validity period.
+
+| Field | Type                                      | Required | Description                                                 |
+|-------|-------------------------------------------|----------|-------------------------------------------------------------|
+| `udv` | `String`                                  | Yes      | UCAN Semantic Version (`1.0.0-rc.1`)                        |
+| `iss` | `DID`                                     | Yes      | Issuer DID (sender)                                         |
+| `aud` | `DID`                                     | Yes      | Audience DID (receiver)                                     |
+| `nbf` | `Integer` (53-bits[^js-num-size])         | No       | "Not before" UTC Unix Timestamp in seconds (valid from)     |
+| `exp` | `Integer \| null` (53-bits[^js-num-size]) | Yes      | Expiration UTC Unix Timestamp in seconds (valid until)      |
+| `nnc` | `String`                                  | Yes      | Nonce                                                       |
+| `mta` | `{String : Any}`                          | No       | [Meta] (asserted, signed data) — is not delegated authority |
+| `sub` | `DID`                                     | Yes      | Principal that the chain is about (the [Subject])           |
+| `can` | `String`                                  | Yes      | [Ability]                                                   |
+| `iff` | `[Caveat]`                                | Yes      | [Caveat]s                                                   |
+
+The payload MUST be serialized as [IPLD] and [signed over][Envelope]. The RECOMMENDED IPLD codec is [DAG-CBOR].
+
+## 3.1 Version
+
+The `udv` field sets the version of the UCAN Delegation specification used in the payload.
```

Comment:
i think i know what you mean, but can you give a quick example to illustrate what you'd like?

Reply:
> i think i know what you mean, but can you give a quick example to illustrate what you'd like?

I was referring to what IPLD Schema calls [keyed unions](https://ipld.io/docs/schemas/using/authoring-guide/#keyed) which later got named "capsule types". What we have in this spec is:

```ts
{
   udv: "1.0.0",
   ...ucan
}
```

If your code can take ☝️ or some other type it becomes hard to guess which one you get. What I was suggesting instead is following:

```
{ "ucan@1.0.0": ucan }
```

Now you it's is clear that it is UCAN of specific version, better yet you can extend that type into a union with more versions you need to support and maybe even other things like invocations, revocations etc...

Downside is languages like TS or Rust aren't really fond of that pattern and you always have to wrap, while in previous pattern if you are aware what is it from context you don't have to wrap, but if you don't you could do something like `{ ucan: { udv: "1.0.0", ...ucan } }`

Reply:
Rust is actually pretty happy with capsule types. We've switched to them in WNFS/rs-wnfs, and it's made some things easier.

However, putting a semantic version inside the key *would* be somewhat harder to deal with, since then you'd like to match on "ucan@1.0.x" or "ucan@1.x.x" in some cases. Putting the version inside is what we do in WNFS, but again that becomes less neat when you want have different fields for different versions. You *could* put the major version into the key, and the rest into the record, but that get's *very* unconventional fast: `{ "ucan@v1": { "subversion": "0.0", ... } }`.